Formula/g/glib.rb

```diff
@@ -56,9 +56,17 @@ def install
     # Avoid the sandbox violation when an empty directory is created outside of the formula prefix.
     inreplace "gio/meson.build", "install_emptydir(glib_giomodulesdir)", ""
 
-    venv = virtualenv_create(libexec, "python3.12")
-    venv.pip_install resources
-    ENV.prepend_path "PYTHONPATH", venv.site_packages
+    # We don't use a venv as virtualenv_create runs `ENV.refurbish_args`. This causes `gint64`
+    # to be detected as `long` rather than `long long` on macOS which mismatches with `int64_t`.
```

Comment:
How does arg refurbishment cause this?

Reply:
Refurbishment shouldn't happen for `configure` scripts but we don't detect non-autotools scripts yet (like meson). Not sure how easy it would be to detect `meson setup` etc too.

Alternatively (or in addition), virtualenv_create affecting other parts of the build also seems like a bug.

Reply:
I think that is correct. Some `-Werror` never make it to Clang after refurbishing.

https://gitlab.gnome.org/GNOME/glib/-/blob/2.80.0/meson.build?ref_type=tags#L1565-1600
* build/meson-logs/meson-log.txt
    ```
    Command line: `clang /private/tmp/glib-20240321-16672-mwccyj/glib-2.80.0/build/meson-private/tmpcl17e3e_/testfile.c -o /private/tmp/glib-20240321-16672-mwccyj/glib-2.80.0/build/meson-private/tmpcl17e3e_/output.obj -c -O0 -Werror=implicit-function-declaration -Werror=unknown-warning-option -Werror=unused-command-line-argument -Werror=ignored-optimization-argument -std=gnu99` -> 0
    ```
* meson.cc
    ```
    clang called with: /private/tmp/glib-20240321-16672-mwccyj/glib-2.80.0/build/meson-private/tmpni3_w87x/testfile.c -o /private/tmp/glib-20240321-16672-mwccyj/glib-2.80.0/build/meson-private/tmpni3_w87x/output.exe -O0 -Werror=implicit-function-declaration
    ```

---

`ENV.refurbish_args` in virtualenv has been around a while now since https://github.com/Homebrew/brew/commit/209d9e8a6eaad3b554f825dbefbfa7e37e20e006. I don't know exact context of why it was added so didn't touch `brew` side.

Reply:
> Not sure how easy it would be to detect `meson setup` etc too.

Could probably handle it with a `meson` and/or `cmake` shim. `meson` should probably be easier; IIRC calling `meson` without an argument for an implicit `meson setup` is deprecated and will be an error soon-ish.

Reply:
FWIW, if `-Werror=incompatible-pointer-types` is _not_ removed by `ENV.refurbish_args` the GLib patch mentioned in https://github.com/libvips/libvips/issues/3901#issuecomment-2002393604 might also fix this (I could upstream that if necessary).